src/pymor/reductors/neural_network.py

```diff
@@ -466,17 +546,53 @@ def compute_training_data(self):
             for datum in self.training_set:
                 if not self.fom:
                     mu, output = datum
-                    sample = self._compute_sample(mu, output=output)
+                    samples = self._compute_sample(mu, output=output)
                 else:
-                    sample = self._compute_sample(datum)
-                self._update_scaling_parameters(sample)
-                self.training_data.extend(sample)
+                    samples = self._compute_sample(datum)
+                for sample in samples:
+                    self._update_scaling_parameters(sample)
+                self.training_data.extend(samples)
+
+    def compute_validation_data(self):
+        """Compute the training samples (the outputs to the parameters of the training set)."""
+        with self.logger.block('Computing validation samples ...'):
+
+            self.validation_data = []
+            for datum in self.validation_set:
+                if not self.fom:
+                    mu, output = datum
+                    samples = self._compute_sample(mu, output=output)
+                else:
+                    samples = self._compute_sample(datum)
+                for sample in samples:
+                    self._update_scaling_parameters(sample)
+                self.validation_data.extend(samples)
+
+    def compute_reduced_basis(self):
+        """empty function to avoid computing a reduced basis."""
+
+    def compute_training_snapshots(self):
+        """empty function to avoid computing training_snapshots."""
+
+    def compute_validation_snapshots(self):
+        """empty function to avoid computing validation_snapshots."""
 
     def _compute_sample(self, mu, output=None):
         """Transform parameter and corresponding output to tensors."""
         if output:
-            return [(mu, output.flatten())]
-        return [(mu, self.fom.output(mu).flatten())]
+            output_size = len(output)
+        else:
+            output = self.fom.output(mu).flatten()
+
+        if not self.is_stationary:
+            output_size = output.shape[0]
+            # conditional expression to check for instationary solution to return self.nt solutions
+            parameters = [mu.with_(t=t) for t in np.linspace(0, self.T, output_size)] if output_size > 1 else [mu]
+            samples = [(param, np.array([out])) for param, out in zip(parameters, output)]
+        else:
+            samples = [(mu, output)]
+        return samples
```

Comment:
I don't understand this function at all. I think we have to be careful here. The output for a single mu might be vector-valued and time-dependent.

This definitely has to be adjusted.